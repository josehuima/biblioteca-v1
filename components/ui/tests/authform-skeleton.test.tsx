import { render } from "@testing-library/react";
import AuthFormSkeleton from "../authform-skeleton"; // Adjust import path as needed
import React from "react";
import { describe, expect, it } from "vitest";
describe("AuthFormSkeleton", () => {
  it("renders correctly for signup", () => {
    const { container } = render(<AuthFormSkeleton type="signup" />);

    // Check if the first section is rendered
    expect(container.querySelector("section")).toBeInTheDocument();

    // Adjusted selector to match a height class with `h-`
    expect(container.querySelector('div[class*="h-"]')).toBeInTheDocument();

    // Additional checks for specific elements
    expect(container.querySelector('div[class*="bg-"]')).toBeInTheDocument(); // Check for background color
  });

  it("renders correctly for signin", () => {
    const { container } = render(<AuthFormSkeleton type="signin" />);

    // Check if the first section is rendered
    expect(container.querySelector("section")).toBeInTheDocument();

    // Adjusted selector to match a height class with `h-`
    expect(container.querySelector('div[class*="h-"]')).toBeInTheDocument();

    // Additional checks for specific elements
    expect(container.querySelector('div[class*="bg-"]')).toBeInTheDocument(); // Check for background color
  });
});
