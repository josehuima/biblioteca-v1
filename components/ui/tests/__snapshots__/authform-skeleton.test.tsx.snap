// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthFormSkeleton > renders correctly for signin 1`] = `
<div>
  <div>
    <div
      class="h-[480px] w-[400px] border-[#ffffff22] bg-slate-950 border-2 pt-8 rounded-[0.75rem] justify-center items-center"
    >
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[24px] w-[160px] rounded"
          data-slot="skeleton"
        />
        <div
          class="bg-accent animate-pulse h-[18px] w-[250px] mt-2 rounded"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[30px] w-[320px] rounded mt-6"
          data-slot="skeleton"
        />
        <div
          class="bg-accent animate-pulse h-[10px] w-[320px] mt-8 rounded"
          data-slot="skeleton"
        />
      </section>
      <section
        class="items-center justify-center mt-8 ml-[40px]"
      >
        <div
          class="bg-accent animate-pulse h-[18px] w-[61px] rounded"
          data-slot="skeleton"
        />
        <div
          class="bg-accent animate-pulse h-[30px] w-[320px] mt-2 rounded"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[30px] w-[320px] rounded mt-6"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[18px] w-[250px] rounded mt-13"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center mt-6"
      >
        <div
          class="bg-accent animate-pulse h-[42px] w-[183px] rounded"
          data-slot="skeleton"
        />
      </section>
    </div>
  </div>
</div>
`;

exports[`AuthFormSkeleton > renders correctly for signup 1`] = `
<div>
  <div>
    <div
      class="h-[640px] w-[400px] border-[#ffffff22] bg-slate-950 border-2 pt-8 rounded-[0.75rem] justify-center items-center"
    >
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[24px] w-[160px] rounded"
          data-slot="skeleton"
        />
        <div
          class="bg-accent animate-pulse h-[16px] w-[250px] mt-2 rounded"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[30px] w-[320px] rounded mt-6"
          data-slot="skeleton"
        />
        <div
          class="bg-accent animate-pulse h-[10px] w-[320px] mt-8 rounded"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-row items-center justify-center space-x-4 mt-9"
      >
        <section
          class="items-center justify-center"
        >
          <div
            class="bg-accent animate-pulse h-[18px] w-[61px] rounded"
            data-slot="skeleton"
          />
          <div
            class="bg-accent animate-pulse h-[30px] w-[152px] mt-2 rounded"
            data-slot="skeleton"
          />
        </section>
        <section
          class="items-center justify-center"
        >
          <div
            class="bg-accent animate-pulse h-[18px] w-[61px] rounded"
            data-slot="skeleton"
          />
          <div
            class="bg-accent animate-pulse h-[30px] w-[152px] mt-2 rounded"
            data-slot="skeleton"
          />
        </section>
      </section>
      <section
        class="items-center justify-center mt-4 ml-[40px]"
      >
        <div
          class="bg-accent animate-pulse h-[18px] w-[61px] rounded"
          data-slot="skeleton"
        />
        <div
          class="bg-accent animate-pulse h-[30px] w-[320px] mt-2 rounded"
          data-slot="skeleton"
        />
      </section>
      <section
        class="items-center justify-center mt-4 ml-[40px]"
      >
        <div
          class="bg-accent animate-pulse h-[18px] w-[61px] rounded"
          data-slot="skeleton"
        />
        <div
          class="bg-accent animate-pulse h-[30px] w-[320px] mt-2 rounded"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[30px] w-[320px] rounded mt-6"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center"
      >
        <div
          class="bg-accent animate-pulse h-[18px] w-[250px] rounded mt-15"
          data-slot="skeleton"
        />
      </section>
      <section
        class="flex flex-col items-center justify-center mt-6"
      >
        <div
          class="bg-accent animate-pulse h-[42px] w-[183px] rounded"
          data-slot="skeleton"
        />
      </section>
    </div>
  </div>
</div>
`;
